src/components/Auth/Signup.jsx

```diff
@@ -0,0 +1,209 @@
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import {
+  Form, FormGroup, Label, Input,
+} from 'reactstrap';
+import { Link } from 'react-router-dom';
+import Side from './Side';
+import img from '../../assets/developers_2_A1_Rectangle_58_pattern.png';
+import { userRegister, registerLoad } from '../../redux/action-creators/signup';
+
+class Register extends Component {
+  state = {
+    username: '',
+    password1: '',
+    password2: '',
+    email: '',
+
+  };
+
+  onChange = (e) => {
+    this.setState({ [e.target.name]: e.target.value });
+  }
+
+  onSubmit = (e) => {
+    e.preventDefault();
+    const {
+      registerSubmit, history, errors,
+    } = this.props;
+    const {
+      username, email, password1, password2,
+    } = this.state;
+
+    if (username.length < 1) {
+      this.setState({ usernameError: 'Username is Required' });
+    }
+
+    if (email.length < 1) {
+      this.setState({ emailError: 'Email is not Valid' });
+    }
+
+    if (password1.length < 1) {
+      this.setState({
+        passwordError: 'Password is Required',
+      });
+    }
+
+    if (password1.length < 8 && password1.length > 1) {
+      this.setState({
+        passwordError: 'Password has to be more than 8 characters',
+      });
+    }
+
+    if (password1 !== password2) {
+      this.setState({
+        passwordError: 'Passwords doesn\'t match',
+      });
+    }
+    if (errors[0] === 'user with this username already exists') {
+      this.setState({
+        usernameError: 'user with this username already exists',
+      });
+    }
+
+    if (errors[0] === 'user with this email already exists') {
+      this.setState({
+        emailError: 'user with this email already exists',
+      });
+    }
+    registerSubmit({ username, email, password: password1 }).then((res) => {
+      if (res.payload.status === 201) {
+        history.push('/');
+      }
+    });
+  }
+
+  render() {
+    const {
+      username,
+      password1,
+      password2,
+      email,
+      usernameError,
+      emailError,
+      passwordError,
+    } = this.state;
+    return (
+      <Fragment>
+        <div className="wrapper">
+          <section className="col-12 col-lg-8 col-md-10 col-sm-10">
+            <div className="container">
+              <div className="row">
+                <div className="col-12 col-md-10 col-lg-7 signup-container text-center">
+                  <Side img={img} />
+                </div>
+                <div className="col-12 col-md-10 col-lg-5 sign-up">
+                  <Form>
+                    <h3 className="sign-up-title mt-4 text-center">
+                    sign
+                      <span>up</span>
+                    </h3>
+                    <FormGroup>
+                      <Label className="label-color">username</Label>
+                      <Input className="input-color" type="username" name="username" placeholder="snow" onChange={this.onChange} value={username} />
```

Comment:
Adding the `required` in input can reduce the code written when checking fields that should not be empty. 
For example: 
```html
<input name="email" placeholder="Input your email" type="email" required/>
```
This can actually can make code from line `34 to 64` absolete.

Reply:
Thank you @me-x-mi I have realized that I am using `Input` component from reactstrap and doesn't support this type of validation.